calculator/parser.py

```diff
@@ -40,49 +50,94 @@ def binary_operator(lhs, rhs, op):
         )
 
 
-def push(e, stack):
+def is_numeric_token(token):
     """
-    Pushes the element (e) to the given stack, which mutates the stack
+    Says whether or not the token input is a number
 
+    Parameter
+    ---------
+    token : token
+        The token to be evalueated
+
+    Returns
+    -------
+    bool: Whether or not this token is a numeric character
+    """
+
+    return type(token) == float or type(token) == int
+
+
+def push(value, stack):
+    """
+    Pushes the element (value) to the given stack, which mutates the stack
     Parameters
     ----------
-    e : element
+    value : element
         The element to be pushed on top of the stack
-
     stack : list
         The stack that e is going to be pushed onto
-
     """
-    stack.append(e)
+    stack.append(value)
 
 
 def pop(stack):
     """
     Retrieves the last element from the given stack and deletes it, which mutates it
-
-    Parameters
     ----------
     stack : list
         The stack (in list form) to be operated on
     """
     return stack.pop()
 
 
-def is_numeric_token(token):
+def peek(stack):
     """
-    Says whether or not the token input is a number
-
+    Returns the top value of the given stack
     Parameter
     ---------
-    token : token
-        The token to be evalueated
-
+    list : list
+        The stack (in list form) to operated on
     Returns
     -------
-    bool: Whether or not this token is a numeric character
+    token : The top token of the stack or queue
```

Comment:
This is just a peek for a stack, not a queue.  Please clarify.

Reply:
This is still left unclear.